refinery/ui/source/js/dashboard/partials/data-sets-card.html

```diff
@@ -0,0 +1,233 @@
+          <span
+            class="refinery-base btn btn-default btn-sm"
+            id="btn-append-to-single-button"
+            uib-dropdown-toggle
+            ng-disabled="disabled">
+            <span ng-if="$ctrl.groupFilter.selectedName !== 'All'">
+              <i class="fa fa-check"></i>
+            </span>
+            Groups <i class="fa fa-caret-down"></i>
+          </span>
+          <ul class="dropdown-menu radio-drop-down" uib-dropdown-menu role="menu"
+            aria-labelledby="btn-append-to-single-button">
+            <li role="menuitem">
+              <label>
+                <input type="radio"
+                  ng-model="$ctrl.groupFilter.selectedName"
+                  ng-change="$ctrl.filterDataSets(0)"
+                  value="All">
+                  All
+              </label>
+            </li>
+            <li ng-repeat="group in $ctrl.groups" role="menuitem">
+               <span ng-if="group.name !== 'Public'">
+                  <label>
+                    <input type="radio"
+                      ng-model="$ctrl.groupFilter.selectedName"
+                      ng-change="$ctrl.filterDataSets(group.id)"
+                      value="{{ group.name }}">
+                    {{ group.name }}
+                  </label>
+               </span>
+            </li>
+          </ul>
+        </div>
+        <div
+          class="refinery-base btn btn-default btn-sm m-l-1"
+          ng-model="$ctrl.groupFilter['public']"
+          ng-change="$ctrl.filterDataSets('public')"
+          uib-btn-checkbox>
+          <span ng-if="$ctrl.groupFilter.public">
+            <i class="fa fa-check"></i>
+          </span>
+          Public
+        </div>
+      </div>
+    </div>
+    <div
+      class="refinery-sub-panel-notice">
+      <div
+        ng-messages="metadataSearch.searchQuery.$error"
+        role="alert">
+        <div ng-message="minlength" class="text-warning notice">
+          <i class="fa fa-warning" aria-hidden="true"></i>
+          <strong>Too short!</strong>: Enter at least 2 characters to search.
+        </div>
+        <div
+          ng-show="!$ctrl.loadingDataSets && $ctrl.dataSets.length === 0 &&
+          $ctrl.searchQueryDataSets
+          .length > 1 && !$ctrl.dataSetsError"
+          class="text-warning notice">
+          <i class="fa fa-warning" aria-hidden="true"></i>
+          <strong>Info</strong>: No data sets found.
+        </div>
+         <div
+          ng-show="$ctrl.loadingDataSets && $ctrl.dataSets.length === 0 && $ctrl
+          .searchQueryDataSets.length === 0 && !$ctrl.dataSetsError"
+          class="text-info notice">
+          <strong>Info</strong>: Loading Data Sets.
+        </div>
+        <div
+          ng-show="!$ctrl.loadingDataSets && $ctrl.dataSets.length === 0 &&
+          $ctrl.searchQueryDataSets.length === 0 && !$ctrl.dataSetsError"
+          class="text-info notice">
+          <strong>Info</strong>: No data sets available.
+        </div>
```

Comment:
Would it be worth pulling out `$ctrl.dataSets.length === 0` and putting it around all three?